coffee_core/src/coffee.rs

```diff
@@ -201,6 +202,17 @@ impl CoffeeManager {
         self.rpc = Some(rpc);
         let path = self.config.cln_config_path.clone();
         let path = path.ok_or_else(|| error!("cln config path not found."))?;
+
+        // The path is the path of the cln config file (eg. ~/.lightning/bitcoin/config)
+        // We need to check that the root folder (eg. ~/.lightning/bitcoin) exists
+        // otherwise we return an error.
+
+        // We can unwrap here because we are sure that the path ends with /config
+        let lightning_path = path.strip_suffix("/config").unwrap();
```

Comment:
If you need to strip this, this means that you are checking in the wrong place.

The check need to be done before we build the path with the `/config`